cli/flox/src/utils/search.rs

```diff
@@ -87,108 +87,148 @@ pub(crate) fn construct_search_params(
-            if inputs.remove(&d.input) {
-                deduped_display_items.push(d.clone());
-            }
-            if inputs.is_empty() {
-                package_to_inputs.remove(&d.package);
+        // For any package that comes from more than one input, mark it as needing to be joined
+        for d in display_items.iter_mut() {
+            if let Some(inputs) = package_to_inputs.get(&d.rel_path) {
+                d.render_with_input = inputs.len() > 1;
             }
         }
     }
-
-    deduped_display_items
-}
-
-/// An intermediate representation of a search result used for rendering
-#[derive(Debug, PartialEq, Clone)]
-struct DisplayItem {
-    /// The input that the package came from
-    input: String,
-    /// The displayable part of the package's attribute path
-    package: String,
-    /// The package description
-    description: Option<String>,
-    /// Whether to join the `input` and `package` fields with a separator when rendering
-    render_with_input: bool,
 }
 
+///
 pub struct DisplaySearchResults {
     /// original search term
     search_term: String,
     /// deduplicated and disambiguated search results
-    deduped_display_items: Vec<DisplayItem>,
+    display_items: DisplayItems,
     /// reported number of results
     count: Option<u64>,
     /// number of actual results (including duplicates)
     n_results: u64,
 }
 
+/// A struct that wraps the functionality needed to print [SearchResults] to a
+/// user.
+impl DisplaySearchResults {
+    /// Display a list of search results for a given search term
+    /// This function is responsible for disambiguating search results
+    /// and printing them to stdout in a user-friendly table-ish format.
+    ///
+    /// If no results are found, this function will print nothing
+    /// it's the caller's responsibility to print a message,
+    /// or error if no results are found.
+    pub(crate) fn from_search_results(
+        search_term: &str,
+        search_results: SearchResults,
+    ) -> Result<DisplaySearchResults> {
+        let n_results = search_results.results.len();
+
+        let display_items = DisplayItems::from_search_results(search_results.results);
+
+        Ok(DisplaySearchResults {
+            search_term: search_term.to_string(),
+            display_items,
+            count: search_results.count,
+            n_results: n_results as u64,
+        })
+    }
+}
+
 impl Display for DisplaySearchResults {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let column_width = self
-            .deduped_display_items
+            .display_items
+            .0
             .iter()
-            .map(|d| {
-                if d.render_with_input {
-                    d.input.len() + d.package.len() + SEARCH_INPUT_SEPARATOR.len()
-                } else {
-                    d.package.len()
-                }
-            })
+            .map(|d| d.to_string().len())
             .max()
```

Comment:
suggestion(non-blocking): use itertools' `max_by_key`
```suggestion
            .max_by_key(|d| d.to_string().len())
```

Reply:
That returns the `DisplayItem` not a `usize` - so leave as is?

Reply:
oh, yeah i guess, doesn't make it easier